packages/datastore/src/datastore.ts

```diff
@@ -142,6 +245,37 @@ class Datastore implements IIterable<Table> {
   redo(transactionId: string): Promise<void> {
     throw '';
   }
+
+  /**
+   * An optional handler for broadcasting transactions to peers.
+   */
+  transactionBroadcastHandler: IMessageHandler | null = null;
+
+
+  private _initTransaction(id: string): void {
+    const context = this._context as Private.MutableContext;
+    if (context.inTransaction) {
+      throw new Error(`Already in a transaction: ${this._currentTransactionId}`);
+    }
+    context.inTransaction = true;
+    context.change = {};
+    context.patch = {};
+    this._currentTransactionId = id;
+  }
+
+  private _finalizeTransaction(): void {
+    const context = this._context as Private.MutableContext;
+    if (!context.inTransaction) {
+      throw new Error('No transaction in progress.');
+    }
+    context.version += 1;
+    context.inTransaction = false;
+  }
+
+  private _tables: {[id: string]: Table<Schema>} = {};
+  private _context: Datastore.Context;
+  private _currentTransactionId: string;
```

Comment:
You could also store the transaction id on the context, which makes more sense to me since that holds transaction-specific data anyway.